jwst/photom/photom.py

```diff
@@ -1097,52 +1098,62 @@ def save_area_info(self, ftab, area_fname):
         if area_fname is not None and area_fname != "N/A":
             pix_area = datamodels.open(area_fname)
 
-            # Copy the pixel area data array to the appropriate attribute
-            # of the science data model
             if self.instrument != 'NIRSPEC':
+                area_ster, area_a2 = None, None
+                try:
+                    # Load the average pixel area values from the AREA reference file header
+                    # Don't need to do this for NIRSpec, because pixel areas have already
+                    # been copied using save_area_nirspec
+                    if isinstance(self.input, datamodels.MultiSlitModel):
+                        # Note that this only copied to the first slit.
+                        self.input.slits[0].area = pix_area.data
+                    else:
+                        ystart = self.input.meta.subarray.ystart - 1
+                        xstart = self.input.meta.subarray.xstart - 1
+                        yend = ystart + self.input.meta.subarray.ysize
+                        xend = xstart + self.input.meta.subarray.xsize
+                        self.input.area = pix_area.data[ystart: yend,
+                                                        xstart: xend]
+                    log.info('Pixel area map copied to output.')
+
+                    area_ster = pix_area.meta.photometry.pixelarea_steradians
+                    if area_ster is None:
+                        log.warning('The PIXAR_SR keyword is missing from %s', area_fname)
+                    area_a2 = pix_area.meta.photometry.pixelarea_arcsecsq
+                    if area_a2 is None:
+                        log.warning('The PIXAR_A2 keyword is missing from %s', area_fname)
+                    log.info('PIXAR_SR and PIXAR_A2 values obtained from AREA reference file.')
```

Comment:
It looks like this info message gets emitted all the time, even if the above few lines issued a warning in the event either keyword value is not found. That could be confusing, because first there'd be a warning message saying the keyword wasn't found and then there'd be a message saying that both of them were obtained from the AREA ref file. So try to put this message into a separate level of if/else block that only gets emitted when both of them are found (i.e. they're both not None).